### DISCUSSION TITLE
What is the recommended way to specify xticks when using plot_series

### DISCUSSION THREAD

**Discussion by ericjb** [alan-turing-institute/sktime#5892] [Q&A]
For the following example, is there an easy (or recommended) way to specify that I want the xticks at all the years (i.e. the Jan.1 locations)? A pointer to an example that does this would be great. (I know how to do this in what seems to me a ridiculously complicated way.)
Thanks
```
y = load_airline()
y.index = y.index.to_timestamp()
fig, ax = plt.subplots()
ax = plot_series(y, markers=[""], ax=ax)
plt.xticks(rotation=90)
plt.show()
```

**Comment by fkiraly** [comment]
hm, this seems like a `matpotlib` question primarily. `plot_series` is `matplotlib` compatible in that it operates on `ax` objects and returns one, the ticks for that you can manipulate with `xticks`.

Here are the official docs from `matplotlib` on datetime ticks: https://matplotlib.org/stable/gallery/text_labels_and_annotations/date.html

The older version has an example with yearly ticks: https://matplotlib.org/3.1.1/gallery/text_labels_and_annotations/date.html

**Reply by ericjb** [reply]
Yes I understand that and I understand the examples you linked to. My problem is that the utility functions that appear in your examples:  
       `ax.xaxis.set_major_locator(years),  ax.xaxis.set_major_formatter(years_fmt),     ax.xaxis.set_minor_locator(months)`

don't work on the ax that is returned by plot_series. I think the problem (for me) originates with the following statements in plot_series

```
xs = [np.argwhere(index.isin(y.index)).ravel() for y in series]
xs_flat = list(flatten(xs))
# set x label of data point to the matching index
def format_fn(tick_val, tick_pos):
    if int(tick_val) in xs_flat:
        return index[int(tick_val)]
    else:
        return ""
# dynamically set x label ticks and spacing from index labels
ax.xaxis.set_major_formatter(FuncFormatter(format_fn))
ax.xaxis.set_major_locator(MaxNLocator(integer=True))
```
These statements have an internal consistency within plot_series itself and date labels are assigned in the correct place. But if I want to change where xticks go on return, the link between the transformed x units and the index dates appears to be lost. (Or is it easy to recover?)

**Reply by fkiraly** [reply]
I see.

Could you paste a full code snippet that fails, as a bug report, where one would expect that it works? Then we can consider it a bug and debug towards it.

An option would be to have a flag to turn off the code you selected - other suggestions for fixes are of course appreciated.

**Comment by ericjb** [comment]
bug report submitted

**Reply by fkiraly** [reply]
https://github.com/sktime/sktime/issues/5895
